### PR TITLE
slowquery慢sql‘查询’

### PR DESCRIPTION
**Pull request by jingyuyinian**
slowquery

### CODE REVIEW

File: contrib/slowquery/slowquery.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    MemoryContextSwitchTo(oldContext); /* Restore the previous memory context */
+}
+
+/* ExecutorEnd hook function to check if a query was slow and log it. */
+static void
+slowquery_ExecutorEnd_hook(QueryDesc *queryDesc) {
+    SlowQueryStackItem *item = slowquery_stack_find(queryDesc->estate->es_query_cxt);
+    if (item != NULL) {
+        TimestampTz endTimestamp = GetCurrentTimestamp(); /* Get the current timestamp */
+        int64 msecsElapsed = DatumGetIntervalP秒差计算函数，计算两个时间戳之间的差值。
```
Comment (WildPigHunter) [inline contrib/slowquery/slowquery.c:84]: Oops~ Is it a un-translated code comment?

File: contrib/slowquery/slowquery.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+static void
+slowquery_stack_free(SlowQueryStackItem *item) {
+    /* Remove the item from the stack and clean up the memory context. */
+    SlowQueryStackItem *current = slowQueryStack;
+    SlowQueryStackItem *prev = NULL;
+    while (current != NULL) {
+        if (current == item) {
+            if (prev != NULL) {
+                prev->next = current->next;
+            } else {
```
Comment (WildPigHunter) [inline contrib/slowquery/slowquery.c:107]: Almost perfect code style, except that the braces and the clause ELSE should take a single line to be consistent with ours.

**Review by WildPigHunter — COMMENTED** [review]
Good Job! It would be better if some test cases could be added.

File: contrib/slowquery/README（slowquery）.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+### 验证方法
+
+在PostgreSQL日志文件中搜索慢查询记录，例如：
+
+```
+LOG: 2023-04-06 10:00:00.000 UTC [慢查询]  SELECT * FROM large_table WHERE some_column IN (SELECT some_id FROM another_table ORDER BY (SELECT pg_sleep(5)));  10s
+```
+
+## 技术架构
+`slowquery`插件通过GUC配置、执行时钩子、内存上下文和日志记录来监控并记录超过指定阈值的慢查询。
```
Comment (WildPigHunter) [inline contrib/slowquery/README（slowquery）.md:77]: Maybe you should translate it into English?

Comment (jingyuyinian) [reply contrib/slowquery/README（slowquery）.md:77]: OK

File: contrib/slowquery/slowquery.c (hunk 2 above)
Comment (jingyuyinian) [reply contrib/slowquery/slowquery.c:107]: 我再做优化

File: contrib/slowquery/slowquery.c (hunk 1 above)
Comment (jingyuyinian) [reply contrib/slowquery/slowquery.c:84]: 已修改

**Pull request merged**